javascript/javascript1/week2/candyhelper.js

```diff
@@ -0,0 +1,50 @@
+//5. Candy helper
+
+const boughtCandyPrices = [];
+const arrCandyPrice = [
+  { candyType: "sweet", price: 0.5 },
+  { candyType: "chocolate", price: 0.7 },
+  { candyType: "toffee", price: 1.1 },
+  { candyType: "chewing-gum", price: 0.03 },
+];
+
+// Function to add candy with its type and weight
+function addCandy(candyType, weight) {
+  boughtCandyPrices.push({ candyType, weight });
+  for (let i = 0; i < boughtCandyPrices.length; i++) {
+    for (let j = 0; j < arrCandyPrice.length; j++) {
+      if (boughtCandyPrices[i].candyType === arrCandyPrice[j].candyType) {
+        boughtCandyPrices[i].candyPrice =
+          boughtCandyPrices[i].weight * arrCandyPrice[j].price;
+      }
```

Comment:
This inner for-loop, can be avoided. But first, good job on learning how to use objects. Next:

If you create you `arrCandyPrice` as follows:
```javascript
const arrCandyPrice = {
  sweet: 0.5,
  chocolate: 0.7,
  toffee: 1.1,
  chewinggum: 0.03,
};
```

then you can acces the price like:
```javascript
const candyName = "sweet"
arrCandyPrice[candyName]
```

meaning the whole inner for loop can be replaced with this lookup instead of iterating over all the elements including the equal operator

Clear?